src/engraving/libmscore/system.cpp

```diff
@@ -2144,4 +2144,19 @@ bool System::hasCrossStaffOrModifiedBeams()
     }
     return false;
 }
+
+//----------------------------------------------------------------
+// yStaffDifference
+// Static utility function. Returns the difference of y-coordinate
+// between any two given staves in any two given systems.
+//----------------------------------------------------------------
+
+double System::yStaffDifference(System* system1, staff_idx_t staffIdx1, System* system2, staff_idx_t staffIdx2)
+{
+    if (!system1 || !system2 || staffIdx1 > system1->staves().size() || staffIdx2 > system2->staves().size()) {
+        // Safety checks
+        return 0.0;
+    }
+    return system1->staff(staffIdx1)->y() - system2->staff(staffIdx2)->y();
```

Comment:
It would be better to get this staff from system1->staves(). So your algorithm would look like this:

```
const std::vector<SysStaff*>& staves1 = system1->staves();
const std::vector<SysStaff*>& staves2 = system2->staves();
if (staffIdx1 >= staves1.size() || staffIdx2 >= staves2.size()) {
    return 0.0;
}

return staves1[staffIdx1]->y() - staves1[staffIdx2]->y();
```

Or you can rewrite it as follows:

```
const Staff* staff1 = system1->staff1(staffIdx1);
const Staff* staff2 = system2->staff2(staffIdx2);

if (!staff1 || !staff2) {
    return 0.0;
}

return staff1->y() - staff2->y()
```